rubrik_cdm/data_management.py

```diff
@@ -666,6 +688,98 @@ def assign_sla(self, object_name, sla_name, object_type, log_backup_frequency_in
 
             return db_sla_lst
 
+        elif object_type == 'oracle_db':
+
+            oracle_db_id = ''
+            db_sla_lst = []
+
+            self.log('Searching the Rubrik cluster for the current Oracle databases.')
+            oracle_db_id = self.object_id(object_name, object_type, hostname=hostname)
+
+            if(oracle_db_id):
+                self.log(
+                    "assign_sla: Determing the SLA Domain currently assigned to the Oracle Database '{}'.".format(object_name))
+
+                oracle_summary = self.get(
+                    'internal',
+                    '/oracle/db/{}'.format(oracle_db_id),
+                    timeout=timeout)
+
+                if (sla_id == oracle_summary['configuredSlaDomainId'] and log_backup_frequency_in_minutes == oracle_summary['logBackupFrequencyInMinutes'] and
+                        log_retention_hours == oracle_summary['logRetentionHours'] and num_channels == oracle_summary['numChannels']):
+                    return "No change required. The Oracle Database '{}' is already assigned to the '{}' SLA Domain with the following log settings:" \
+                            " log_backup_frequency_in_minutes: {}, log_retention_hours: {} and num_channels: {}.".format(
+                                object_name, sla_name, log_backup_frequency_in_minutes, log_retention_hours, num_channels)
+
+                else:
+                    self.log(
+                        "assign_sla: Assigning the Oracle Database '{}' to the '{}' SLA Domain.".format(
+                            object_name, sla_name))
+
+                    config = {}
+                    if log_backup_frequency_in_minutes is not None:
+                        config['logBackupFrequencyInMinutes'] = log_backup_frequency_in_minutes
+                    if log_retention_hours is not None:
+                        config['logRetentionHours'] = log_retention_hours
+                    if num_channels is not None:
+                        config['numChannels'] = num_channels
+
+                    config['configuredSlaDomainId'] = sla_id
+
+                    patch_resp = self.patch("internal", "/oracle/db/{}".format(oracle_db_id), config, timeout)
+                    db_sla_lst.append(patch_resp)
+            else:
+                raise InvalidParameterException(
+                    "Database ID not found for instance '{}'").format(object_name)
+            
+            return db_sla_lst
```

Comment:
Whats the story around `db_sla_lst` here?

Can we just return the PATCH right away instead of appending?

Reply:
@drew-russell pushed an update to address this